js/test/Exchange/test.sharedMethods.js

```diff
@@ -0,0 +1,206 @@
+'use strict'
+
+const assert = require ('assert');
+const Precise = require ('../../base/Precise');
+
+function logTemplate (exchange, method, entry) {
+    return ' <<< ' + exchange.id + ' ' + method + ' ::: ' + exchange.json (entry) + ' >>> ';
+}
+
+function reviseStructureKeys (exchange, method, entry, format, emptyNotAllowedFor = []) {
```

Comment:
I would probably replace `revise` with `assert`, revise is not very clear/standard

Reply:
sure, u'r right, i never meant those method names to be final. we can have assert, check, or whatever you think